### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by potash**
Generates some queries. We can revise API, etc.

### CODE REVIEW

**Comment by mbauman** [comment]
Flake8 is not happy with your coding style.  https://travis-ci.org/dssg/collate/jobs/169018818#L175

I'll look into configuring some of those options.  I'd be fine with, e.g., line lengths under 100 chars.

**Comment by potash** [comment]
:)

File: collate/collate.py [hunk 1]
```diff
@@ -1 +1,144 @@
...
+        in which case the cross product of those is used. If quantity is a
+        collection than name should also be a collection of the same length.
+        """
+        self.quantities = make_list(quantity)
+        self.functions = make_list(function)
+
+        if name is not None:
+            self.quantity_names = make_list(name)
+            if len(self.quantity_names) != len(self.quantities):
+                raise ValueError("Name length doesn't match quantitie length")
```
Comment (thcrock) [inline collate/collate.py]: typo

File: collate/collate.py [hunk 2]
```diff
@@ -1 +1,144 @@
...
+class Aggregate(object):
+    """
+    An object representing one or more SQL aggregate columns in a groupby
+    """
+    def __init__(self, quantity, function, name=None):
+        """
+        Args:
+            quantity: an SQL string expression for the quantity to aggregate
+            function: an SQL aggregate function
+            name: a name for the quantity, usd in the aggregate column name
```
Comment (mbauman) [inline collate/collate.py]: `used`

**Pull request merged**